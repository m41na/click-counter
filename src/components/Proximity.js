import React from 'react';

export default ({increment, decrement}) => {
    return (
        <section className="clicker">
            <button type="button" onClick={increment}>++</button>
            <button type="button" onClick={decrement}>--</button>
        </section>
    );
}