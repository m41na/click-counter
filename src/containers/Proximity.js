import {connect} from 'react-redux';
import {incrAction, decrAction} from '../actions';
import Proximity from '../components/Proximity';

const mapStateToProps = state => ({
    value: state.distance.value
});

const mapDispatchToProps = dispatch => ({
    increment: () => {
        dispatch(incrAction())
    },
    decrement: () => {
        dispatch(decrAction())
    }
});

export default connect(mapStateToProps, mapDispatchToProps)(Proximity);