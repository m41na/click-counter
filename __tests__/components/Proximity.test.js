import React from 'react';
import { shallow, mount, render } from 'enzyme';
import Proximity from '#/components/Proximity';

const initialProps = {
    distance: {'value': 100},
    increment(){
        console.log('increment');
    },
    decrement(){
        console.log('decrement');
    }
};

describe('AwayView behavior', () => {
    describe('render()', () => {        
        it('should render the component with two buttons', () => {
            const component = shallow(<Proximity {...initialProps} />);
            expect(component).toMatchSnapshot();        

            //assert list has two tasks
            const list = component.find("button");
            expect(list).toHaveLength(2);
        });
    });
});