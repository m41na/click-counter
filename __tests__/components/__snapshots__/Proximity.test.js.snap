// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AwayView behavior render() should render the component with two buttons 1`] = `
<section
  className="clicker"
>
  <button
    onClick={[Function]}
    type="button"
  >
    ++
  </button>
  <button
    onClick={[Function]}
    type="button"
  >
    --
  </button>
</section>
`;
